lib/foreman_tasks_core/runner/command_runner.rb

```diff
@@ -23,18 +23,25 @@ def refresh
         end
       end
 
+      # @override
+      # by default, we pass the null input stream to the sub-process, as
+      # we don't expect to run in interactive mode. Override if different behaviour
+      # is expected
+      def input_stream
+        @input_stream ||= File.open(File::NULL, "r")
+      end
+
       def close
         close_io
       end
 
       private
 
       def close_io
-        @command_out.close if @command_out && !@command_out.closed?
-        @command_out = nil
-
-        @command_in.close if @command_in && !@command_in.closed?
-        @command_in = nil
+        [@command_out, @command_in, @input_stream].each do |stream|
+          stream.close if stream && !stream.closed?
+          stream = nil
```

Comment:
This modifies the local binding, not the instance variable

Reply:
Nice catch, updated